Tools/scripts/build_options.py

```diff
@@ -413,7 +421,7 @@ def __init__(self,
     Feature('Filesystem', 'FILESYSTEM_SYS', 'AP_FILESYSTEM_SYS_ENABLED', 'Enable @SYS/ filesystem', 0, None),
     Feature('Filesystem', 'APJ_TOOL_PARAMETERS', 'FORCE_APJ_DEFAULT_PARAMETERS', 'Enable apj_tool parameter area', 0, None),
 
-    Feature('Networking', 'PPP Support', 'AP_NETWORKING_BACKEND_PPP', 'Enable PPP networking', 0, None),
+    Feature('Networking', 'PPP Support', 'AP_NETWORKING_BACKEND_PPP', 'Enable PPP networking', 0, None, "enable-ppp"),
```

Comment:
Could we put these hacks back in `waf`, please?

Should just be a dictionary mapping from A to B?

Reply:
it needs to be here to keep backward compatibility with existing documented options like --enable-ppp. We won't need it for new options. It would be an awful lot uglier to have this in wscript

Reply:
if we're willing to break existing documented configure options we could remove these

Reply:
Or throw a warning in 4.6 that the option is deprecated and remove in 4.7?

Reply:
> if we're willing to break existing documented configure options we could remove these

I'm thinking that breaking them would be ok, although it will be a small inconvenience initially for a very small minority who can probably deal with it, because in the long term, part of the power of this feature comes from it's consistency - everything will work the same way.

Reply:
Surely 
`Feature('Networking', 'PPP Support', 'AP_NETWORKING_BACKEND_PPP', 'Enable PPP networking', 0, None, "enable-ppp"),`
should be "PPP" not "PPP Support" - it seems to be the only one with a space in the label field.

Reply:
If we want to support --enable-ppp, could you not _also_ support --enable-PPP, ditto for ekf2/EKF2, ekf3/EKF3.

I'm thinking that someone might want to build a script that would generate a ./waf configure command line. These special cases would make that difficult, but supporting both (deprecating the lowercase option) might be a good interim solution.